src/lightning/pytorch/plugins/precision/double.py

```diff
@@ -12,28 +12,76 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from contextlib import contextmanager
-from typing import Any, cast, Generator, List, Literal, Tuple
+from typing import Any, Generator, Literal
 
 import torch
 import torch.nn as nn
 from lightning_utilities.core.apply_func import apply_to_collection
 from torch import Tensor
-from torch.optim import Optimizer
 
 import lightning.pytorch as pl
 from lightning.fabric.plugins.precision.utils import _convert_fp_tensor
-from lightning.pytorch.overrides.base import _LightningPrecisionModuleWrapperBase
+from lightning.fabric.utilities.device_dtype_mixin import _DeviceDtypeModuleMixin
 from lightning.pytorch.plugins.precision.precision_plugin import PrecisionPlugin
+from lightning.pytorch.utilities.rank_zero import rank_zero_deprecation
 
 
-class LightningDoublePrecisionModule(_LightningPrecisionModuleWrapperBase):
+class DoublePrecisionPlugin(PrecisionPlugin):
+    """Plugin for training with double (``torch.float64``) precision."""
+
+    precision: Literal["64-true"] = "64-true"
+
+    def convert_module(self, module: nn.Module) -> nn.Module:
+        return module.double()
+
+    @contextmanager
+    def init_context(self) -> Generator[None, None, None]:
+        """A context manager to change the default tensor type when initializing module parameters or tensors.
+
+        See: :meth:`torch.set_default_dtype`
+        """
+        default_dtype = torch.get_default_dtype()
+        torch.set_default_dtype(torch.float64)
+        yield
+        torch.set_default_dtype(default_dtype)
+
+    @contextmanager
+    def forward_context(self) -> Generator[None, None, None]:
+        """A context manager to change the default tensor type.
+
+        See: :meth:`torch.set_default_dtype`
+        """
+        default_dtype = torch.get_default_dtype()
+        torch.set_default_dtype(torch.float64)
+        yield
+        torch.set_default_dtype(default_dtype)
+
+    def convert_input(self, data: Any) -> Any:
+        return apply_to_collection(data, function=_convert_fp_tensor, dtype=Tensor, dst_type=torch.double)
+
+
+class LightningDoublePrecisionModule(_DeviceDtypeModuleMixin, nn.Module):
     """LightningModule wrapper which converts incoming floating point data in ``*_step`` and ``forward`` to double
     (``torch.float64``) precision.
 
+    Deprecated. Use :meth:`~lightning.pytorch.core.hooks.ModelHooks.configure_model` instead.
```

Comment:
```suggestion
    .. deprecated:: Use :meth:`~lightning.pytorch.core.hooks.ModelHooks.configure_model` instead.
```
see: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated